crates/red_knot_python_semantic/resources/mdtest/type_properties/is_subtype_of.md

```diff
@@ -0,0 +1,232 @@
+static_assert(not is_subtype_of(Literal[1], Intersection[int, Not[1]]))
+```
+
+## Class literal types
+
+```py
+from abc import ABC, ABCMeta
+from types import ModuleType
+from typing import _SpecialForm
+from typing_extensions import Literal, LiteralString
+import typing
+from knot_extensions import TypeOf, is_subtype_of, static_assert
+
+static_assert(is_subtype_of(TypeOf[bool], type[int]))
+static_assert(is_subtype_of(TypeOf[int], TypeOf[int]))
+static_assert(is_subtype_of(TypeOf[int], object))
+
+static_assert(is_subtype_of(TypeOf[Literal], _SpecialForm))
+static_assert(is_subtype_of(TypeOf[Literal], object))
+
+static_assert(is_subtype_of(TypeOf[ABC], ABCMeta))
+static_assert(is_subtype_of(ABCMeta, type[object]))
+
+static_assert(is_subtype_of(tuple[int], tuple))
+static_assert(is_subtype_of(TypeOf[str], type))
+
+static_assert(is_subtype_of(TypeOf[typing], ModuleType))
+static_assert(is_subtype_of(TypeOf[1:2:3], slice))
+
+static_assert(not is_subtype_of(type, type[int]))
+static_assert(not is_subtype_of(TypeOf[str], type[Any]))
+static_assert(not is_subtype_of(type[str], TypeOf[str]))
+
+static_assert(not is_subtype_of(TypeOf[int], TypeOf[object]))
+static_assert(not is_subtype_of(TypeOf[int], int))
+
+static_assert(not is_subtype_of(_SpecialForm, TypeOf[Literal]))
+static_assert(not is_subtype_of(ABCMeta, type[type]))
+```
+
+## `AlwaysTruthy` and `AlwaysFalsy`
+
+```py
+from typing_extensions import Never
+from knot_extensions import AlwaysTruthy, AlwaysFalsy, is_subtype_of, static_assert
+
+static_assert(is_subtype_of(Literal[1], AlwaysTruthy))
+static_assert(is_subtype_of(Literal[0], AlwaysFalsy))
+
+static_assert(is_subtype_of(AlwaysTruthy, object))
+static_assert(is_subtype_of(AlwaysFalsy, object))
+
+static_assert(is_subtype_of(Never, AlwaysTruthy))
+static_assert(is_subtype_of(Never, AlwaysFalsy))
+
+static_assert(not is_subtype_of(Literal[1], AlwaysFalsy))
+static_assert(not is_subtype_of(Literal[0], AlwaysTruthy))
+
+static_assert(not is_subtype_of(str, AlwaysTruthy))
+static_assert(not is_subtype_of(str, AlwaysFalsy))
+```
+
+## User-defined classes
+
+```py path="unions.py"
+from knot_extensions import TypeOf, is_subtype_of, static_assert
+
+class Base: ...
+class Derived(Base): ...
+class Unrelated: ...
+
+reveal_type(Base)  # revealed: Literal[Base]
+reveal_type(Derived)  # revealed: Literal[Derived]
+
+static_assert(is_subtype_of(TypeOf[Base], type))
+static_assert(is_subtype_of(TypeOf[Base], object))
+
+static_assert(is_subtype_of(TypeOf[Base], type[Base]))
+static_assert(is_subtype_of(TypeOf[Derived], type[Base]))
+static_assert(is_subtype_of(TypeOf[Derived], type[Derived]))
+
+static_assert(not is_subtype_of(TypeOf[Base], type[Derived]))
+static_assert(is_subtype_of(type[Derived], type[Base]))
+
+def _(flag: bool):
+    U = Base if flag else Unrelated
+
+    reveal_type(U)  # revealed: Literal[Base, Unrelated]
+
+    static_assert(is_subtype_of(TypeOf[U], type))
+    static_assert(is_subtype_of(TypeOf[U], object))
```

Comment:
Using the type aliases above, this could probably also be written without a `flag` helper?

```suggestion
static_assert(is_subtype_of(TypeOf[LiteralBase | LiteralUnrelated], type))
static_assert(is_subtype_of(TypeOf[LiteralBase | LiteralUnrelated], object))
```